client/src/lib.rs

```diff
@@ -518,8 +550,17 @@ pub trait ServerGatewayApis {
     fn get_options(&self) -> &ServerGatewayOptions;
 }
 
+impl WorkflowServiceClientTrait for ServerGateway {
+    // TODO: Will implement 1:1 generated client methods. Boilerplate bonanza
+}
+
+impl WorkflowServiceClientTrait for NamespacedServerGateway {
+    // TODO: Will implement 1:1 generated client methods. Boilerplate bonanza
+}
```

Comment:
probably not the same trait because the namespaced one doesn't (or at least shouldn't) accept a namespace.

Reply:
This is the raw trait, they both must implement it. The methods will be exactly as in the generated client.

Reply:
The namespaced client wouldn't have the same API though..

Reply:
It won't. It will also implement the higher level trait (which is mentioned in the notes). This is just Rust trait stuff. It needs to implement the lower level one because the higher level one depends on it.

Reply:
Arguably the `NamespacedServerGateway` could just have a reference to the underlying `ServerGateway` in its struct instead of impl'ing the same trait.

Reply:
It does, and the traits could just be totally separate and if you want the raw thing you just have to call the accessor first. Perhaps that's a better direction.